c++/src/kj/compat/http.c++

```diff
@@ -2637,6 +2637,18 @@ public:
 
   uint64_t receivedByteCount() override { return receivedBytes; }
 
+  kj::Maybe<kj::String> getPreferredExtensions(ExtensionsContext ctx) override {
+    KJ_IF_MAYBE(config, compressionConfig) {
+      if (ctx == ExtensionsContext::RESPONSE) {
```

Comment:
Should we go ahead and have this method return null in the case that the context suggests a reverse of direction? E.g. pumping from one client to another, rather than proxying client-to-server? optimizedPump won't work in that case anyway so there's no reason to specify a preference.

Reply:
I'm having some trouble seeing how we could know this information from within this method. I'd think the caller would need to know ahead of time whether it was trying to get the client-to-server config, or whether it was trying to pump from one client to another, no?

Reply:
You can tell based on the presence or absence of `maskKeyGenerator` whether the WebSocket object is the client side vs. the server side of a connection. See `tryPumpFrom()`, where it will refuse to do an optimized pump unless exactly one of the two sides has a `maskKeyGenerator`. Since optimized pumping will be refused in that case anyway, we should return null here.